fil_fungible_token/src/token/mod.rs

```diff
@@ -278,8 +285,380 @@ where
         // - receiver hook should see the new balances...
 
         // if all succeeded, atomically commit the transaction
-        state.save(&self.bs)?;
-
+        self.save_state(&state)?;
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use cid::Cid;
+    use fvm_shared::econ::TokenAmount;
+    use num_traits::Zero;
+
+    use crate::blockstore::SharedMemoryBlockstore;
+
+    use super::Token;
+
+    #[test]
+    fn it_instantiates() {
+        let state_cid = Cid::default();
+        let mut token = Token::new(SharedMemoryBlockstore::new(), state_cid);
```

Comment:
This is weird constructing with an invalid state CID. I think maybe we can find better construction patterns. Instead of `new`, I suggest adding `create()`, which creates a new state and returns a `Token` referencing it, and then only using `new` with a valid CID (but it probably won't actually check).

A bigger change would hold the `TokenState` member in the `Token`, and then we would rename `new` to `load` and it would actually enforce correct internal state. I'm unsure whether that's worthwhile tho.